server/application/application.go

```diff
@@ -291,14 +314,23 @@ func (s *Server) queryRepoServer(ctx context.Context, a *v1alpha1.Application, a
 
 // GetManifests returns application manifests
 func (s *Server) GetManifests(ctx context.Context, q *application.ApplicationManifestQuery) (*apiclient.ManifestResponse, error) {
-	a, err := s.appLister.Get(*q.Name)
+	if q.Name == nil || *q.Name == "" {
+		return nil, fmt.Errorf("invalid request: application name is missing")
+	}
+	appName := q.GetName()
+	appNs := s.appNamespaceOrDefault(q.GetAppNamespace())
+	a, err := s.appLister.Applications(appNs).Get(appName)
```

Comment:
Should we do the "is allowed" check before the `GET`? Just to help lighten unnecessary load on k8s

Reply:
I was thinking a little bit about this.

My original intent was to have the check for allowed namespaces being done after the RBAC check, so as to not allow low-privileged users to enumerate the allowed namespaces easily by trial & error.

Optimally, even the K8s API call would be made _after_ the RBAC check, but since we need the application's project information in order to perform the RBAC check, this does not seem to be possible.

Maybe we can go for the following approach:

1. Request is received
1. Perform first "superficial" RBAC check without checking for the project being valid (e.g. RBAC resource `foo/*` would match any application name, likewise, `*/foobar` would match an Application named `foobar`). Error out early if RBAC doesn't match.
1. Check for allowed namespace
1. Perform K8s API call to retrieve the application
1. Perform another RBAC check, this time considering the Application's project

WDYT? This would be out-of-scope for this PR, but maybe worth another improvement PR.

Reply:
> Optimally, even the K8s API call would be made after the RBAC check, but since we need the application's project information in order to perform the RBAC check, this does not seem to be possible.

Personally I think we should make a breaking API change and force all calls to include the project name. I started using this pattern with the `/terminal` endpoint. But that's for another day.  :-) 

> Maybe we can go for the following approach:

Agreed, that would be a good, non-breaking approach!